shared/src/commonMain/kotlin/co/touchlab/kampkit/models/BreedModel.kt

```diff
@@ -26,40 +30,80 @@ class BreedModel : KoinComponent {
+        val stale = isBreedListStale(currentTimeMS)
+        val networkBreedDataState: DataState<ItemDataSummary>
+        if (stale) {
+            networkBreedDataState = getBreedsFromNetwork(currentTimeMS)
+            when (networkBreedDataState) {
+                DataState.Empty -> {
+                    // Pass the empty response along
+                    emit(networkBreedDataState)
+                }
+                is DataState.Success -> {
+                    dbHelper.insertBreeds(networkBreedDataState.data.allItems.map { it.name })
+                }
+                is DataState.Error -> {
+                    // Pass the error along
+                    emit(networkBreedDataState)
+                }
+
+                DataState.Loading -> {
+                    // Won't ever happen
+                }
+            }
+        }
+
+        getBreedsFromCache().conflate().collect {
+            emit(it)
+        }
+    }
+
+    fun getBreedsFromCache(): Flow<DataState<ItemDataSummary>> =
         dbHelper.selectAllItems()
             .map { itemList ->
                 log.v { "Select all query dirtied" }
-                ItemDataSummary(
-                    itemList.maxByOrNull { it.name.length },
-                    itemList
+                DataState.Success(
+                    ItemDataSummary(
+                        itemList.maxByOrNull { it.name.length },
+                        itemList
+                    )
                 )
             }
 
-    suspend fun getBreedsFromNetwork(): String? {
-        fun isBreedListStale(currentTimeMS: Long): Boolean {
-            val lastDownloadTimeMS = settings.getLong(DB_TIMESTAMP_KEY, 0)
-            val oneHourMS = 60 * 60 * 1000
-            return (lastDownloadTimeMS + oneHourMS < currentTimeMS)
+    private fun isBreedListStale(currentTimeMS: Long): Boolean {
+        val lastDownloadTimeMS = settings.getLong(DB_TIMESTAMP_KEY, 0)
+        val oneHourMS = 60 * 60 * 1000
+        val stale = lastDownloadTimeMS + oneHourMS < currentTimeMS
+        if (!stale) {
+            log.i { "Breeds not fetched from network. Recently updated" }
         }
+        return stale
+    }
 
-        val currentTimeMS = Clock.System.now().toEpochMilliseconds()
-        if (isBreedListStale(currentTimeMS)) {
-            try {
-                val breedResult = ktorApi.getJsonFromApi()
-                log.v { "Breed network result: ${breedResult.status}" }
-                val breedList = breedResult.message.keys.toList()
-                log.v { "Fetched ${breedList.size} breeds from network" }
-                dbHelper.insertBreeds(breedList)
-                settings.putLong(DB_TIMESTAMP_KEY, currentTimeMS)
-            } catch (e: Exception) {
-                log.e(e) { "Error downloading breed list" }
-                return "Unable to download breed list"
+    suspend fun getBreedsFromNetwork(currentTimeMS: Long): DataState<ItemDataSummary> {
```

Comment:
I'm not convinced that it makes sense for this function to return a `DataState`. It only gets called by `getBreeds()` which then packs its own `DataState` depending on the response here.